src/main/java/net/minecraftforge/common/extensions/IForgeBlockEntity.java

```diff
@@ -189,4 +196,69 @@ default void requestModelDataUpdate()
      {
          return EmptyModelData.INSTANCE;
      }
+
+    /**
+     * Marks this block entity as having been removed from the level for the given reason.
+     *
+     * @param reason the reason for the block entity's removal
+     */
+     void setRemoved(RemovalReason reason);
+
+    /**
+     * Reasons for the removal of a {@link BlockEntity} from the level.
+     */
+    enum RemovalReason
```

Comment:
I'm not sure all these cases, and the `boolean chunk` parameter, are useful. Wouldn't REMOVED and UNLOADED be enough, possibly UNKNOWN too?

Reply:
The various removal reasons were added in an effort to provide as much information to the modder as reasonably possible. We certainly could cut down the amount of values in this enum (though for two values, it'd be better then to use a `boolean`), but I felt at the time that providing the info would be of some potential use and wasn't too hard to implement at least.

Reply:
More info means more cases to handle for the modder though. I think that most use cases would be covered by a simple boolean `wasUnloaded`, probably with a default of false.

Reply:
I am inclined to agree that at least adding a second boolean to `RemovalReason` for removed (and not coming back) vs just unloaded would make sense and be useful for devs (`REMOVED` and `REPLACED_IN_WORLD`, not sure about `UNKNOWN` and I haven't looked into how `CLEARED_CHUNK` is used)

Reply:
@pupnewfster: That sounds like a reasonable idea to me, so modders can rely on the boolean instead of hardcoding the individual reasons (and possibly needing to add any further new reasons should they come up). This way, we retain the information provided by the individual `RemovalReason`s (which was my original reason for the enum) and provide a simple boolean check for modders to use (as @Technici4n notes).